utils/lister.hh

```diff
@@ -91,6 +91,20 @@ public:
         return scan_dir(std::move(dir), std::move(type), do_show_hidden, std::move(walker), [] (const fs::path& parent_dir, const directory_entry& entry) { return true; });
     }
 
+    /** Overloads accepting std::string_view as the first parameter */
+    static future<> scan_dir(std::string_view dir, dir_entry_types type, show_hidden do_show_hidden, walker_type walker, filter_type filter) {
```

Comment:
why not just use the overload of `scan_dir(fs::path...)`? IMHO, the overloads accepting `sstring` is not necessary.

Reply:
That's a good point

Reply:
I checked the PR introducing overloads for `seastar::sstring` and back then, `lister` was based on `using path = boost::filesystem::path`. Not going into what the API of it was, right now cppreference comments on the constructors:

> Constructs the path from a character sequence (format interpreted as specified by `fmt`) provided by source (4, 5), which is a pointer or an input iterator to a null-terminated character/wide character sequence, an `std::basic_string` or an `std::basic_string_view`, or represented as a pair of input iterators `[first, last)`.

However, it follows up with

> (5) and (7) participate in overload resolution only if `Source` and `path` are not the same type, and either:
>  Source is a specialization of `std::basic_string` or `std::basic_string_view`, or
>  `std::iterator_traits<std::decay_t<Source>>::value_type` is valid and denotes a possibly `const`-qualified encoding character type (`char`, `char8_t`, `char16_t`, `char32_t`, or `wchar_t`).

so `seastar::sstring` should be a valid overload. I'll test it on my machine and if everything works fine, I'll edit the PR.